packages/patternfly-4/react-core/src/components/Pagination/PaginationOptionsMenu.tsx

```diff
@@ -89,16 +96,31 @@ export class PaginationOptionsMenu extends React.Component<PaginationOptionsMenu
     });
   };
 
+  handleNewPerPage = (_evt: React.MouseEvent | React.KeyboardEvent | MouseEvent, newPerPage: number) => {
+    const { page, onPerPageSelect, itemCount, defaultToFullPage } = this.props;
+    let newPage = page;
+    if (defaultToFullPage) {
+      if (itemCount / newPerPage !== newPage) {
+        while (newPage && (itemCount - (newPerPage * newPage)) < 0) {
```

Comment:
We might want to tweak this logic to handle the case where the user selects a number to display per page that's greater than the entire dataset. For example, I have 6 items, but specify 50 per page. In this case, I'd think to just display all the available rows we do have.

![2019-11-05 15 16 24](https://user-images.githubusercontent.com/5942899/68243394-685a0d00-ffe0-11e9-8b9c-1f3ac9ad8afd.gif)

Otherwise, this seems to work great, nice work!